test/kwok/kwok.test.sh

```diff
@@ -54,6 +54,32 @@ function test_node_ready() {
   fi
 }
 
+# Check pod status
+function test_check_pod_status() {
+  for row in $(kubectl get po -ojson | jq -r '.items[] | @base64'); do
+      _jq() {
+      echo ${row} | base64 --decode | jq -r ${1}
+      }
+    pod_name=$(_jq '.metadata.name')
+    node_name=$(_jq '.spec.nodeName')
+    host_network=$(_jq '.spec.hostNetwork')
+    host_ip=$(_jq '.status.hostIP')
+    pod_ip=$(_jq '.status.podIP')
+    node_ip=$(kubectl get nodes ${node_name} -ojson|jq '.status.addresses[]| select( .type == "InternalIP" )|.address'|tr -d '"')
+    if [[ $host_ip != $node_ip ]]; then
+        echo "Error: ${pod_name} Pod: HostIP (${host_ip}) is not equal to the IP on its Node (${node_ip})"
+        return 1
+    fi
+    if [[ $hostNetwork == "true" ]]; then
```

Comment:
```suggestion
    if [[ $host_network == "true" ]]; then
```

Reply:
Nice catch. This is critical.

Reply:
Applied the changes [here](https://github.com/kubernetes-sigs/kwok/compare/6937431a58b0a2149d9fe50882102895d9b55002..4849934ad12e13eb52cffd10cc563bf79e04f519).